modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/LimitedWriteRateFileIO.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.persistence.file;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
+import org.apache.ignite.IgniteInterruptedException;
+import org.apache.ignite.internal.IgniteInterruptedCheckedException;
+import org.apache.ignite.internal.util.BasicRateLimiter;
+
+/**
+ * File I/O providing the ability to limit the write rate.
+ * <p>
+ * Note: read operations are not limited, only write and transfer operations are limited.
+ */
+public class LimitedWriteRateFileIO extends FileIODecorator {
```

Comment:
It might be worth trying to override the `writeFully` methods and just call them in the `write`. It more clearly depicts what is really going on here.

Reply:
Done, thanks!